File: server/app/services/telegram/agent-manager.js
const { Pool } = require('pg'); // Assuming you use 'pg' library
const pool = new Pool({ connectionString: process.env.DATABASE_URL });

async function getAiResponse(chatId, userText) {
  // 1. DATABASE LOOKUP: Find which business/agent this chat belongs to
  // For MSMEs, you likely store which bot_token or business_id maps to this chat
  const res = await pool.query('SELECT agent_config FROM active_agents WHERE telegram_chat_id = $1', [chatId]);
  const agentConfig = res.rows[0]?.agent_config || "Default assistant";

  // 2. AI CALL: (Example using placeholder for your LLM of choice)
  // const reply = await openAI.createChatCompletion({...});
  const reply = `AI Agent Response for: ${userText}`; // Replace with real AI call

  // 3. INSIGHTS: Log to Postgres for your Dashboard
  await pool.query(
    'INSERT INTO logs (chat_id, message, direction, sentiment) VALUES ($1, $2, $3, $4)',
    [chatId, userText, 'inbound', 'neutral']
  );

  return reply;
}

module.exports = { getAiResponse };

File: server/app/services/telegram/agent_manager.py
import asyncpg
import os

DATABASE_URL = os.getenv("DATABASE_URL")

async def get_agent_response(chat_id: int, user_text: str):
    # 1. Connect to Postgres
    conn = await asyncpg.connect(DATABASE_URL)
    try:
        # 2. Fetch business context (How should this agent behave?)
        # row = await conn.fetchrow("SELECT context FROM agents WHERE chat_id=$1", chatId)
        
        # 3. AI Logic (Placeholder for OpenAI/Claude)
        ai_reply = f"Agent Response to: {user_text}" 

        # 4. Log for Dashboard Insights
        await conn.execute(
            "INSERT INTO chat_logs (chat_id, message, direction, created_at) VALUES ($1, $2, $3, NOW())",
            chat_id, user_text, 'inbound'
        )
        return ai_reply
    finally:
        await conn.close()